src/sentry/sentry_metrics/metrics_interface.py

```diff
@@ -0,0 +1,42 @@
+from abc import ABC, abstractmethod
+from typing import Mapping
+
+from sentry.sentry_metrics.use_case_id_registry import UseCaseID
+
+
+class GenericMetricsBackend(ABC):
+    @abstractmethod
+    def counter(
+        self,
+        use_case_id: UseCaseID,
+        org_id: int,
+        project_id: int,
+        metric_name: str,
+        value: int,
+        tags: Mapping[str, str],
+    ) -> None:
+        raise NotImplementedError()
+
+    @abstractmethod
+    def set(
+        self,
+        use_case_id: UseCaseID,
+        org_id: int,
+        project_id: int,
+        metric_name: str,
+        value: int,
```

Comment:
I think this should take a `Sequence[int]` (if the caller already has a few values, we shouldn't make them call set multiple times)

Reply:
I don't really have any strong opinion about this but i'm more accustomed to the statsd interface where you only pass one value and not a sequence https://github.com/statsd/statsd/blob/master/docs/metric_types.md

Reply:
Also, should we allow strings and other types in here? Why only an int?